### PR TITLE
Fix #2825: Removed unused private field

### LINKED ISSUE #2825: Remove the unused private field
- [ ] Remove this unused "spec" private field

https://github.com/fabric8io/kubernetes-client/blob/5df8dae7fa31b009fe8971bacb50dccd63a47b6d/crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceInfo.java#L34

[Sonarcloud.io issue page](https://sonarcloud.io/project/issues?fileUuids=AXeKpnxFmR8wXoZXwwEO&id=fabric8io_kubernetes-client&open=AXeKppG0mR8wXoZXwwF9&resolved=false)

### PR DESCRIPTION
**Pull request by gandalf0151**
## Description

- Removed unused private field "spec" 
Fixes #2825 
## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by gandalf0151** [comment]
@rohanKanojia : Kindly take a look at the PR
Also, for some reason, sonar scanner test has failed after connection timeout. Could you please restart that test ?

**Comment by rohanKanojia** [comment]
Umm, I'll restart the job

**Comment by rohanKanojia** [comment]
I am curious about your motivation behind contributing to our project. Are you contributing because of Gsoc? We are not really sure if we will be participating in Gsoc this year.

**Review by oscerd — APPROVED** [review]

**Comment by gandalf0151** [comment]
@oscerd : Thanks for reviewing the PR
@rohanKanojia : I'm a CS Undergrad in my junior year. I started to contribute in fabric8io for GSOC. Although fabric8io willn't be taking part in GSOC, still, I'll keep contributing. 
I was looking for orgs which deal with Java technology and devops. Could you direct me towards any other orgs which involve the same ? 
Thanks

File: crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java [hunk 1]
```diff
@@ -115,18 +115,15 @@ private CustomResourceInfo add(TypeElement element) {
...
-      final CustomResourceInfo info = new CustomResourceInfo(element, spec, status, processingEnv);
+      final CustomResourceInfo info = new CustomResourceInfo(element, status, processingEnv);
       System.out.println(
         "Generating '"
           + info.crdName()
           + "' version '"
           + info.version()
           + "' with "
           + crClassName
-          + " (spec: "
```
Comment (metacosm) [inline crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java:128]: The field is indeed not used but might be at some point to inform the CRD generation. However, even if the field is not used, the logging that informs the users about which CRD is being generated from which CR shouldn't be removed. So, at the very least, this needs to stay.

**Review by metacosm — CHANGES_REQUESTED** [review]

File: crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java (hunk 1 above)
Comment (gandalf0151) [reply crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java]: @metacosm : Sorry for the late reply
I added the logging field. Please take a look. Thanks

**Comment by gandalf0151** [comment]
I guess I'll have to close this PR as the code for the above files have changed and it'll take a considerable amount of time to resolve conflicts. 
Also, the issue #2825 doesn't seem to exist in sonarcloud anymore.
Do correct me if I'm wrong.

**Pull request closed without merging**